File: detekt/wrapper/src/testFixtures/java/io/buildfoundation/bazel/detekt/execute/TestDetekt.java
package io.buildfoundation.bazel.detekt.execute;

import java.io.PrintStream;

class TestDetekt implements Detekt {

    enum ExecuteResult {
        Success, Failure
    }

    private final ExecuteResult executeResult;

    TestDetekt(ExecuteResult executeResult) {
        this.executeResult = executeResult;
    }

    @Override
    public void execute(String[] args, PrintStream output, PrintStream error) {
        if (executeResult == ExecuteResult.Failure) {
            throw new RuntimeException();
        }
    }
}

File: detekt/wrapper/src/testFixtures/kotlin/io/buildfoundation/bazel/detekt/execute/TestDetekt.kt
package io.buildfoundation.bazel.detekt.execute

import java.io.PrintStream

internal class TestDetekt(private val executionResult: ExecutionResult) : Detekt {

    enum class ExecutionResult {
        Success, Failure
    }

    override fun execute(args: Array<String>, outputPrinter: PrintStream, errorPrinter: PrintStream) {
        if (executionResult == ExecutionResult.Failure) {
            throw RuntimeException()
        }
    }
}